Assets/BossRoom/Scripts/Shared/Net/UnityRelayUtilities.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Threading.Tasks;
+using UnityEngine;
+using Unity.Services.Relay;
+using Unity.Services.Relay.Allocations;
+using Unity.Services.Relay.Models;
+
+namespace BossRoom
+{
+    public static class RelayJoinCodeThing
+    {
+        public static string RelayJoinCode = String.Empty;
+    }
+    public class RelayUtility
+    {
+        async public static
+            Task<(string ipv4address, ushort port, byte[] allocationIdBytes, byte[] connectionData, byte[] key, string
+                joinCode)> AllocateRelayServerAndGetJoinCode(int maxConnections, string region = null)
+        {
+            Response<AllocateResponseBody> allocationResponse;
+            Response<JoinCodeResponseBody> createJoinCodeResponse;
+            try
+            {
+                allocationResponse =
+                    await RelayService.AllocationsApiClient.CreateAllocationAsync(
+                        new CreateAllocationRequest(new AllocationRequest(maxConnections)));
+            }
+            catch
+            {
+                Debug.LogError("Relay create allocation request failed");
+                throw;
+            }
+
+            var allocation = allocationResponse.Result.Data.Allocation;
+
+            Debug.Log($"server: {allocation.ConnectionData[0]} {allocation.ConnectionData[1]}");
+            Debug.Log($"server: {allocation.AllocationId}");
+
+            try
+            {
+                createJoinCodeResponse = await RelayService.AllocationsApiClient.CreateJoincodeAsync(
+                    new CreateJoincodeRequest(
+                        new JoinCodeRequest(allocationResponse.Result.Data.Allocation.AllocationId)));
+            }
+            catch
+            {
+                Debug.LogError("Relay create join code request failed");
+                throw;
+            }
+
+            return (allocation.RelayServer.IpV4, (ushort) allocation.RelayServer.Port, allocation.AllocationIdBytes,
+                allocation.ConnectionData, allocation.Key, createJoinCodeResponse.Result.Data.JoinCode);
+        }
+
+        async public static
+            Task<(string ipv4address, ushort port, byte[] allocationIdBytes, byte[] connectionData, byte[]
+                hostConnectionData, byte[] key)> JoinRelayServerFromJoinCode(string joinCode)
```

Comment:
That util seems very useful, shouldn't it be part of Unity Relay SDK?

Reply:
same for the other one, why aren't these in the Relay SDK?

Reply:
Because the relay SDK is not Unity Specific

Reply:
ah I see, makes sense for Join
for allocation, it could make sense to have something does both "create allocation" and "get join code"? But that's mostly nitpicking at this point, now that the API is simpler, it seems fine